### PR TITLE
[SEDONA-293] Implement ST_IsCollection

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-293. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- I implemented the ST_IsCollection function.

## How was this patch tested?

- In my local Ubuntu VM environment.

## Did this PR include necessary documentation updates?

- Yes, I have added the necessary information to the readmes.

### CODE REVIEW

File: docs/api/flink/Function.md [hunk 1]
```diff
@@ -1087,6 +1087,20 @@ Example:
...
+
+Introduction: Returns `TRUE` if the argument type is a geometry collection type.
+
+Format: `ST_IsCollection(geom: geometry)`
+
+Since: `v1.5.0`
+
+Example:
+
+```sql
```
Comment (jiayuasu) [inline docs/api/flink/Function.md:1100]: Please provide two examples of the output. One is `true` and another one is `false`

File: docs/api/sql/Function.md [hunk 2]
```diff
@@ -1082,6 +1082,20 @@ SELECT ST_IsClosed(ST_GeomFromText('LINESTRING(0 0, 1 1, 1 0)'))
...
+Introduction: Returns `TRUE` if the argument type is a geometry collection type.
+
+Format: `ST_IsCollection(geom: geometry)`
+
+Since: `v1.5.0`
+
+Example:
+
+```sql
+SELECT ST_IsCollection(ST_GeomFromText('MULTIPOINT(0 0), (6 6)'))
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:1096]: Same here

File: flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java [hunk 3]
```diff
@@ -13,6 +13,7 @@
  */
 package org.apache.sedona.flink.expressions;
 
+import org.apache.calcite.interpreter.Scalar;
```
Comment (jiayuasu) [inline flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java:16]: Why is the calcite class introduced here?

File: common/src/main/java/org/apache/sedona/common/Functions.java [hunk 4]
```diff
@@ -988,6 +988,14 @@ public static double frechetDistance(Geometry g1, Geometry g2) {
         return GeomUtils.getFrechetDistance(g1, g2);
     }
 
+    public static boolean isCollection(Geometry geometry) {
+        String geoType = geometry.getGeometryType();
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/Functions.java:992]: Please implement test cases in Sedona common

File: sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala [hunk 5]
```diff
@@ -384,5 +384,9 @@ object st_functions extends DataFrameAPI {
 
   def ST_CoordDim(geometry: String): Column = wrapExpression[ST_CoordDim](geometry)
 
+  def ST_IsCollection(geometry: Column): Column = wrapExpression[ST_IsCollection](geometry)
```
Comment (jiayuasu) [inline sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala:387]: Spark DataFrame style API is not tested. Please add test cases.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java (hunk 3 above)
Comment (furqaankhan) [reply flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java:16]: I will remove it, sorry will take care next time.

File: common/src/main/java/org/apache/sedona/common/Functions.java (hunk 4 above)
Comment (jiayuasu) [reply common/src/main/java/org/apache/sedona/common/Functions.java:992]: Yes

File: docs/api/sql/Function.md [hunk 6]
```diff
@@ -1082,6 +1082,29 @@ SELECT ST_IsClosed(ST_GeomFromText('LINESTRING(0 0, 1 1, 1 0)'))
 
 Output: `false`
 
+## ST_IsCollection
+
+Introduction: Returns `TRUE` if the argument type is a geometry collection type.
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:1087]: The doc here is not comprehensive. Returns TRUE if the geometry type of the input is a geometry collection type. Collection types are the following:

GEOMETRYCOLLECTION

MULTI{POINT,POLYGON,LINESTRING}

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**